docs/experimental/viz_points_kde.py

```diff
@@ -0,0 +1,356 @@
+
+    texture = Texture()
+
+    cmap = (255*colormap).astype(np.uint8)
+    cmap = rgb_to_vtk(cmap)
+
+    texture.SetInputDataObject(cmap)
+    texture.SetWrap(Texture.ClampToEdge)
+    texture.SetInterpolate(interpolate)
+    texture.MipmapOn()
+    texture.SetBlendingMode(0)
+
+    target_actor.GetProperty().SetTexture(texture_name, texture)
+
+
+def shader_custom_uniforms(actor : actor.Actor, shader_type : str):
+    """Eases the passing of uniform values to the shaders by returning ``actor.GetShaderProperty().GetVertexCustomUniforms()``,
+    that give access to the ``SetUniform`` methods.
+    Parameters
+    ----------
+    actor : actor.Actor
+          Actor which the uniform values will be passed to.
+    shader_type : str
+          Shader type of the uniform values to be passed. It can be:
+          * "vertex"
+          * "fragment"
+          * "geometry"
+    """
+    if shader_type == "vertex":
+        return actor.GetShaderProperty().GetVertexCustomUniforms()
+    elif shader_type == "fragment":
+        return actor.GetShaderProperty().GetFragmentCustomUniforms()
+    elif shader_type == "geometry":
+        return actor.GetShaderProperty().GetGeometryCustomUniforms()
+    else:
+        raise ValueError("Shader type unknown.")
+
+
+def normalize(array : np.array, min : float = 0.0, max : float = 1.0, axis : int = 0):
+    """Converts an array to a given desired range.
+
+    Parameters
+    ----------
+    array : np.ndarray
+    Array to be normalized.
+    min : float
+    Bottom value of the interval of normalization. If no value is given, it is passed as 0.0.
+    max : float
+    Upper value of the interval of normalization. If no value is given, it is passed as 1.0.
+
+    Returns
+    -------
+
+    array : np.array
+    Array converted to the given desired range.
+    """
+    if np.max(array) != np.min(array):
+        return ((array - np.min(array))/(np.max(array) - np.min(array)))*(max - min) + min
+    else:
+        raise ValueError(
+            "Can't normalize an array which maximum and minimum value are the same.")
+
+
+kde_dec = """
+float kde(vec3 point, float sigma){
+    return exp(-1.0*pow(length(point), 2.0)/(2.0*sigma*sigma) );
+}
+"""
+
+kde_impl = """
+float current_kde = kde(normalizedVertexMCVSOutput, sigma);
+color = vec3(current_kde);
+fragOutput0 = vec4(color, 1.0);
+"""
+
+tex_dec = """
+const float normalizingFactor = 1.0; // Parameter to be better defined later
+
+vec3 color_mapping(float intensity, float normalizingFactor){
+    float normalizedIntensity = intensity/normalizingFactor;
+    return texture(colormapTexture, vec2(normalizedIntensity,0)).rgb;
+}
+"""
+
+tex_impl = """
+vec2 renorm_tex = normalizedVertexMCVSOutput.xy*0.5 + 0.5;
+float intensity = texture(screenTexture, renorm_tex).r;
+
+if(intensity<=0.0){
+    discard;
+}else{
+    color = color_mapping(intensity, normalizingFactor).rgb;
+    fragOutput0 = vec4(color, 1.0);
+}
+"""
+
+
+fs_dec = compose_shader([kde_dec])
```

Comment:
This is not necessary since it is just one string, you can pass to the billboard actor `kde_dec` and `kde_impl` directly.